### PR TITLE
ZCS-8834:QuarantineDevice API

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
Added new API QuarantineDevice to replace SuspendDevice 
Deprecated SuspendDevice

### CODE REVIEW

File: soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceRequest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import com.zimbra.soap.type.AccountSelector;
+
+/**
+ * @zm-api-command-deprecation-info Note: to be deprecated in Zimbra 9. Use zimbraAdmin QuarantineDevice instead.
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required true
+ * @zm-api-command-description Suspend a device or all devices attached to an account from further sync actions
+ */
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name=SyncAdminConstants.E_SUSPEND_DEVICE_REQUEST)
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceRequest.java]: This should be quarantine

File: soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceRequest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+import com.google.common.base.MoreObjects;
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.common.soap.SyncAdminConstants;
+import com.zimbra.common.soap.SyncConstants;
+import com.zimbra.soap.admin.type.DeviceId;
+import com.zimbra.soap.type.AccountSelector;
+
+/**
+ * @zm-api-command-deprecation-info Note: to be deprecated in Zimbra 9. Use zimbraAdmin QuarantineDevice instead.
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceRequest.java]: This is the new one, so this message doesn't belong here.

File: soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceResponse.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.zimbra.common.soap.SyncAdminConstants;
+import com.zimbra.common.soap.SyncConstants;
+import com.zimbra.soap.admin.type.DeviceStatusInfo;
+
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name=SyncAdminConstants.E_SUSPEND_DEVICE_RESPONSE)
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceResponse.java]: This should be quarantine

File: soap/src/java/com/zimbra/soap/admin/message/SuspendDeviceRequest.java [hunk 4]
```diff
@@ -30,6 +30,7 @@
 import com.zimbra.soap.type.AccountSelector;
 
 /**
+ * @zm-api-command-deprecation-info Note: to be deprecated. Use zimbraAdmin QuarantineDevice instead.
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/admin/message/SuspendDeviceRequest.java]: Can probably mention deprecation version.

File: soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceRequest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * @zm-api-command-deprecation-info Note: to be deprecated in Zimbra 9. Use zimbraAccount QuarantineDevice instead.
+ * @zm-api-command-network-edition
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required false
+ * @zm-api-command-description Suspend a device from further sync actions
+ */
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name=SyncConstants.E_SUSPEND_DEVICE_REQUEST)
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceRequest.java]: This should be quarantine

File: soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceRequest.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import com.zimbra.common.soap.SyncConstants;
+import com.zimbra.soap.sync.type.DeviceId;
+
+/**
+ * @zm-api-command-deprecation-info Note: to be deprecated in Zimbra 9. Use zimbraAccount QuarantineDevice instead.
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceRequest.java]: This is the new file, so this doesn't belong here

File: soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceResponse.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import com.zimbra.common.soap.SyncConstants;
+import com.zimbra.soap.sync.type.DeviceStatusInfo;
+
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name=SyncConstants.E_SUSPEND_DEVICE_RESPONSE)
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceResponse.java]: This should be quarantine.

File: soap/src/java/com/zimbra/soap/sync/message/SuspendDeviceRequest.java [hunk 8]
```diff
@@ -27,6 +27,7 @@
 import com.zimbra.soap.sync.type.DeviceId;
 
 /**
+ * @zm-api-command-deprecation-info Note: to be deprecated. Use zimbraAccount QuarantineDevice instead.
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/sync/message/SuspendDeviceRequest.java]: Can mention version

File: soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceRequest.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.common.soap.SyncAdminConstants;
+import com.zimbra.common.soap.SyncConstants;
+import com.zimbra.soap.admin.type.DeviceId;
+import com.zimbra.soap.type.AccountSelector;
+
+/**
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required true
+ * @zm-api-command-description Suspend a device or all devices attached to an account from further sync actions
```
Comment (chinmay15) [inline soap/src/java/com/zimbra/soap/admin/message/QuarantineDeviceRequest.java:35]: suspend -> quarantine
attached to device id

File: soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceRequest.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+import javax.xml.bind.annotation.XmlRootElement;
+
+import com.zimbra.common.soap.SyncConstants;
+import com.zimbra.soap.sync.type.DeviceId;
+
+/**
+ * @zm-api-command-network-edition
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required false
+ * @zm-api-command-description Suspend a device from further sync actions
```
Comment (chinmay15) [inline soap/src/java/com/zimbra/soap/sync/message/QuarantineDeviceRequest.java:33]: suspend -> quarantine

**Pull request merged**